### PR TITLE
Clarify process for proposal adoption.

### PR DESCRIPTION
**Pull request by barath**
This PR clarifies the process for proposal adoption in the WG per the discussion in the CG today.

### CODE REVIEW

**Review by csarven — CHANGES_REQUESTED** [review]
Again, the [W3C Recommendation Track Readiness Best Practices](https://www.w3.org/Guide/standards-track/) (BP) is the document in particular that can be of help here. The current text in the proposed Solid WG charter:

>The Working Group will not adopt new proposals until they have matured through the W3C Solid Community Group or another similar incubation phase.

is derived from the document referenced above:

>The Working Group will not adopt new proposals until they have matured through the Web Platform Incubator Community Group or another similar incubation phase.

As mentioned elsewhere (with links provided), the text is already used in other W3C charters, and there is deliberation to introduce that into the charter template. The text has been acknowledged and used by groups. It is hard to go wrong with that baseline. Hence, I strongly recommend that the current text stays as is for starters.

I find the proposal in this PR inadequate because it goes in the opposite direction, i.e., it weakens what's intended by W3C with respect to incubation, and does not actually address the underlying unclarity that's be raised by few individuals.

As agreed, we can indeed further clarify the process on what constitutes "matured", and so, that actually requires us to focus more on the incubation criteria as mentioned in the BP (see the questions under "Readiness Criteria"), in particular "empirical criteria", and to adopt something similar to [determine when a spec under incubation is ready to propose to a Working Group](https://wicg.github.io/admin/intent-to-migrate.html).

Until that kind of documentation is in place, I believe that the existing text in the WG charter suffices. It is sufficiently clear to active CG participants, and W3C staff and well-versed AC representatives are already familiar with W3C charters, processes, and so forth, for whom the existing text is demonstrably clear. The WG can simply re-use the criteria in the BP to assess readiness.

**Comment by melvincarvalho** [comment]
> The Working Group will not adopt new proposals until they have matured through the Web Platform **Incubator** Community Group or another similar **incubation** phase.

The 'Web Platform Incubator Community Group' emphasizes the term 'incubator'.

Regarding the 'W3C Solid Community Group', I'd like to clarify that it wasn't initiated as an incubator group. Ideally, the WG should have autonomy in maturing specs, with the CG adapting as needed. The revised language captures this sentiment effectively.

**Comment by elf-pavlik** [comment]
Can we add a link to https://www.w3.org/Guide/standards-track/#criteria , maybe by prefixing the sentence currently in the draft with:

"**To ensure that proposals meet the [W3C Readiness Criteria](https://www.w3.org/Guide/standards-track/#criteria), The Working Group will not adopt new proposals until ...**"

**Review by RubenVerborgh — APPROVED** [review]
From an imec perspective, we support the proposed wording. For us, it's crucial that:
- the WG has a clear and independent mandate to work on what is urgent and important for Solid to become a reality;
- the CG has a clearly defined purpose in support of the WG's mandate.

The old phrasing could be misinterpreted as creating a restrictive relationship, which I do not consider desired. In contrast, the new positive phrasing is open-ended, in which the CG gets a clear incubation pathway without creating unneeded obligations or stress on the CG.

Moving forward, imec aims to play a strong role in bot the standardization work within the WG, and incubation work in the CG. The proposed phrasing gives both groups the best chance at success, IMHO.

**Comment by woutermont** [comment]
@melvincarvalho, every W3C CG has incubation as one of its main goals, i.e., as a free-to-participate social environment or the incubation of ideas towards possible future standardization.

While I don't find the first attempt in this PR too bad, to me it feels more vaguer rather than more clear, when compared to the original wording. I.m.o. it introduces two extra uncertainties:

- The rewrite from 'will not' to 'aims to' now seems to leave in the middle whether proposals should be incubated **at all**.
- The removal of of the 'similar' qualifier leads to interpretations that just about any organisation can propose 'incubated' drafts.

---

As I have already listed in a [comment on the previous PR](https://github.com/solid/solid-wg-charter/pull/47#issuecomment-1691118740), there are a number of aspects we definitely want to capture clearly:

- Proposals don't start from scratch in a WG. It accepts them only if their ideas are deemed matured enough to work with. **Proposals need an incubation phase**. To summarize the guide on [Recommandation Track Readiness](https://www.w3.org/Guide/standards-track/): _proposals to a WG have to identifiy a real-world problem, formulate success criteria for a solution, and propose an initial draft of a solution that has reached a well-socialized level of maturity in a community, i.e., there is general consensus that the specification is likely to be implemented and used on the web._

- **The most obvious incubation phase is the CG**: a W3G groups specifically tasked with incubation. Whenever the CG agrees they have a mature enough draft, they can propose it to the WG, which can accept or deny it on their own terms.

- **Other incubation phases are not excluded**. Work from other CGs, or other organized groups of people that work in a democratic and open way similar (_problem, solution criteria, maturity, concensus in community ..._), can just as well propose work items to the WG, on which the latter can agan decide on its own terms.

---

Just throwing out a sentence combining this:

> The Working Group will be open to adopt proposals that fulfill the requirements of [W3C Recommendation Track Readiness](https://www.w3.org/Guide/standards-track/#criteria). In particular, the Working Group will only adopt proposals it deems sufficiently mature, whether proposed through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or another similar incubation phase.

File: charter/index.html [hunk 1]
```diff
@@ -288,7 +288,7 @@ <h3>
             If additional in-scope Recommendation-track deliverables need to be added to the Charter before the Charter expires, the Working Group will prepare an updated Charter that differs only in deliverables.
           </p>
           <p>
-            The Working Group will not adopt new proposals until they have matured through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or another similar incubation phase.
+            The Working Group aims to adopt proposals it deems sufficiently mature; such proposals can be submitted through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or other incubation phase.
```
Comment (csarven) [inline charter/index.html:291]: Pavlik's or Wouter's suggestions [[1](https://github.com/solid/solid-wg-charter/pull/48#issuecomment-1691568434)], [[2](https://github.com/solid/solid-wg-charter/pull/48#issuecomment-1691601128)] both acknowledge prior work, understanding of incubation, and BP at W3C, and would be adequate to approve the PR.

```suggestion
            The Working Group will use the [W3C Recommendation Track Readiness Criteria](https://www.w3.org/Guide/standards-track/#criteria) when evaluating proposals. The Working Group will not adopt new proposals until they have matured through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or another similar incubation phase.
```

**Comment by oolivo** [comment]
I challenge the recommendation that the current text stays as is. Reading the [W3C Recommendation Track Readiness Best Practices](https://www.w3.org/Guide/standards-track/) it is clear that that sentence is used as an example of what one working group does to address incubation concerns, not as a general recommendation of what all WG's should do. It even prefaces this text with:

"The language in the [Web Platform WG charter](http://w3c.github.io/charter-html/group-charter.html#deliverables) describes one community's practice:" 

It's clearly just meant to be an example and not taken as gospel that should not be amended or edited. Furthermore, as Barath mentioned in his original PR, out of 126 existing WG charters, only 3 have language resembling this (including the webplatform charter that is cited). So if we're using the argument of precedents to not amend the text, it's pretty weak. 

I'm OK with the proposed PR as is, but will agree that @woutermont provides a recommendation that is a bit more explicit and clear (and provides cited context), and I'm always for more clarity. @barath would you consider incorporating @woutermont's recommendation into this PR?

**Comment by barath** [comment]
Thank you all for the discussion. I believe a combination of the text proposed in the above discussion is reasonable. I'll make the edit shortly.

File: charter/index.html [hunk 2]
```diff
@@ -288,7 +288,7 @@ <h3>
             If additional in-scope Recommendation-track deliverables need to be added to the Charter before the Charter expires, the Working Group will prepare an updated Charter that differs only in deliverables.
           </p>
           <p>
-            The Working Group will not adopt new proposals until they have matured through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or another similar incubation phase.
+           The Working Group will adopt proposals that fulfill the requirements of <a href="https://www.w3.org/Guide/standards-track/#criteria">W3C Recommendation Track Readiness</a>. In particular, the Working Group will only adopt proposals it deems sufficiently mature. Proposals can be made to the Working Group through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or another incubation phase.
```
Comment (elf-pavlik) [inline charter/index.html:291]:
```suggestion
           The Working Group will adopt proposals that fulfill the requirements of <a href="https://www.w3.org/Guide/standards-track/#criteria">W3C Recommendation Track Readiness</a>. In particular, the Working Group will only adopt proposals it deems sufficiently mature. Proposals can be made to the Working Group through the <a href="https://www.w3.org/groups/cg/solid">W3C Solid Community Group</a> or another similar incubation phase.
```

I think *similar* is essential here. Quoting from https://www.w3.org/Guide/standards-track/#criteria

> This consensus could emerge from an incubation phase in WICG or another CG, or in a WG that has an established culture of taking and vetting suggestions from its public mailing list. 

If *similar* is still to vague we could even quote: "*that has an established culture of taking and vetting suggestions from its public mailing list.*", understanding that public github repositories can be considered comparable to mailing list.

**Review by justinwb — APPROVED** [review]
From the standpoint of Janeiro Digital this text looks reasonable and addresses our concern that legitimate proposals incubated outside of the CG would fail to be considered

File: charter/index.html (hunk 2 above)
Comment (woutermont) [reply charter/index.html:291]: Wholeheartedly agree. "Similar", "comparable" ... in other words: any incubation process achieving the aforementioned readiness criteria (addresses real problem, formulates measureable solution, based on consensus in social community, mature enough to expect implementaitons).

Comment (csarven) [reply charter/index.html:291]: I agree to keep "similar" in as that entails a particular meaning and expectation that's taken in the context of the Readiness Criteria. I don't have a strong opinion on unpacking that further or quoting the BP, but if that's introduced here, that'd be fine too - although it'd be unnecessarily verbose. Charters are almost never intended to be taken raw without the context / references in mind. So, for me, "similar" is sufficient, as it was for other groups as well.

@barath  , can you please incorporate the suggested change to add "similar". Happy to approve the PR based on that.

Comment (pchampin) [reply charter/index.html:291]: I would also rather keep "similiar" in the sentence, as suggested by @elf-pavlik [above](https://github.com/solid/solid-wg-charter/pull/48#discussion_r1304521418)

**Review by pchampin — CHANGES_REQUESTED** [review]

**Comment by melvincarvalho** [comment]
> @melvincarvalho, every W3C CG has incubation as one of its main goals, i.e., as a free-to-participate social environment or the incubation of ideas towards possible future standardization.

Hi @woutermont .  You may or may not know this but at the w3c there's traditionally been three levels of group:

CG -- community groups, generally casual in nature, open to all
XG -- incubator groups, incubate standards track work, reports or a charter, normally invite only
WG -- working group, formal, member only or invited expert, REC track deliverables

Case in point a couple of examples:

From about 2007-2010 we ran the social web incubator group, this produced a [Final Report](https://www.w3.org/2005/Incubator/socialweb/wiki/FinalReport) which was one of the inputs to the creation of both Solid and the Social Web Working Group, that came after.  After that the SWWG the Social web transitioned to a CG

Similarly up until 2010 there was a WebID XG which transitioned to a CG, which is where much of the work was done for solid, in addition to the Read Write Web CG where things like web access control were developed.

At some point the CG and XG system was made one thing, which is under the umbrella of the CG, right now.

When I started the CG, I was asked if we were going to work on specs, and I said it could be in scope.  We appointed two chairs and let the group take it's own course.  Before even Inrupt joined we had dozens of members, so that gave a bit more interest for it to be a place where people could gather.  The idea of the CG was in the style of a diverse community, definitely allowing casual participation, with a more formal subset of those wanting to work on specs related things, for those that wanted it.  So elements of CG and XG emerged over time.  

The WG is REC track work, and can be pulled from the various places in solid's history, the Solid CG, the WebID CG, the RW CG, and the stuff inrupt and others have done with the commercial work.

The key thing for solid rn is to get the WG over the line and then Solid hopefully to REC status.  Hope that explains the history a bit better.

**Review by woutermont — APPROVED** [review]

**Review by csarven — APPROVED** [review]
I'm glad we came full circle after much bikeshedding (and unnecessary detouring) where the end is result is essentially equivalent to was acknowledged with the obvious context when the original text was introduced back in 2023-04. :shrug:

**Comment by woutermont** [comment]
If the end result is clarity for more people than before, I'm glad. 🙂 So thanks everyone for the constructive stance. 

I also want to thank @melvincarvalho for his historical perspective, which indeed confirms that CG's are an obvious place to do incubation.

**Review by pchampin — APPROVED** [review]

**Review by laurensdeb — APPROVED** [review]

**Pull request merged**